### PR TITLE
Add opd only option.

### PR DESCRIPTION
**Pull request by jbkalmbach**

### CODE REVIEW

File: python/lsst/ts/imsim/imsimCmpt.py [hunk 1]
```diff
@@ -518,6 +518,8 @@ def _mapOpdToZk(self, rotOpdInDeg, numOpd):
             else:
                 opdRot = opd
 
+            # opdRot = np.fliplr(opdRot)
```
Comment (suberlak) [inline python/lsst/ts/imsim/imsimCmpt.py:521]: Is this commented out because of the Zernike flips done earlier?

File: python/lsst/ts/imsim/closedLoopTask.py [hunk 2]
```diff
@@ -534,6 +548,10 @@ def _runSim(
             else:
                 self.ofcCalc.set_fwhm_data(fwhm, sensor_id)
 
+            # Flip zernikes that are not symmetric across the y-axis
+            # based upon flip in batoid coordinate system.
+            wfe[:, [1, 4, 6, 9, 11, 12, 14, 16]] *= -1.0
```
Comment (suberlak) [inline python/lsst/ts/imsim/closedLoopTask.py:553]: Perhaps in the PR discussion page it would be worth mentioning where this flip has been investigated?  Eg. a link to a notebook or some other note explaining why is that change needed.

Comment (suberlak) [reply python/lsst/ts/imsim/closedLoopTask.py:553]: I think part of https://sitcomtn-003.lsst.io/ explains the difference between Zemax and phosim - was it that ts_wep  was following Zemax, and Batoid had the z-axis pointing the opposite way  ?

File: python/lsst/ts/imsim/imsimCmpt.py [hunk 3]
```diff
@@ -651,7 +653,7 @@ def mapOpdDataToListOfWfErr(self, opdZkFileName, sensorIdList, sensorNameList):
             sensorWavefrontData = SensorWavefrontError(numOfZk=self.numOfZk)
             sensorWavefrontData.sensorId = sensorId
             sensorWavefrontData.sensorName = sensorName
-            sensorWavefrontData.annularZernikePoly = opdZk[sensorId]
+            sensorWavefrontData.annularZernikePoly = opdZk[sensorId] / 1e3
```
Comment (suberlak) [inline python/lsst/ts/imsim/imsimCmpt.py:656]: I guess this converts nm to um - why was it not there before? Does the addition of OPD only mode mean that one needs to proceed to um?

File: tests/test_imsimCmpt.py [hunk 4]
```diff
@@ -248,7 +248,7 @@ def testMapOpdDataToListOfWfErr(self):
             self.assertEqual(sensorId, mapSensorNameAndId[refSensorName])
 
             zkInWfErr = wfErr.annularZernikePoly
-            delta = np.sum(np.abs(zkInWfErr - opdZk[sensorId]))
+            delta = np.sum(np.abs(zkInWfErr - opdZk[sensorId] / 1e3))
```
Comment (suberlak) [inline tests/test_imsimCmpt.py:251]: Same as above - could you comment why is this unit change happening?

**Review by suberlak — APPROVED** [review]
Looks good, only few comments on being consistent in units and explaining the rotation of Zk part.

File: python/lsst/ts/imsim/imsimCmpt.py (hunk 1 above)
Comment (jbkalmbach) [reply python/lsst/ts/imsim/imsimCmpt.py:521]: Yes, I'll remove this.

File: python/lsst/ts/imsim/imsimCmpt.py (hunk 3 above)
Comment (jbkalmbach) [reply python/lsst/ts/imsim/imsimCmpt.py:656]: Yes,  imSim outputs OPD in nanometers so need to change to microns to be consistent with what WEP expects. Before we didn't need to interface OPD results with WEP because we could just do it when making plots but now we need to feed in OPD to WEP so need to do the conversion here in the code. I'll add a comment on this.

File: tests/test_imsimCmpt.py (hunk 4 above)
Comment (jbkalmbach) [reply tests/test_imsimCmpt.py:251]: Same as above. Will add same comment here.

File: python/lsst/ts/imsim/closedLoopTask.py (hunk 2 above)
Comment (jbkalmbach) [reply python/lsst/ts/imsim/closedLoopTask.py:553]: Slide 10 in this presentation from the 2023 PCW shows the comparison between imSim and Phosim: https://docs.google.com/presentation/d/1jMbH0cm74elBpXStOEZ1A40-P4ctHpiXhgngwJtHpmA/edit?usp=sharing 

There is a flip across the y-axis in the donuts so there is a reflection in Zernikes in the x-direction between the two.

There is additional discussion on this issue here in slack cataloging Josh Meyers' exploration of the issue: https://lsstc.slack.com/archives/CHXKSF3HC/p1691456649673369

Comment (jbkalmbach) [reply python/lsst/ts/imsim/closedLoopTask.py:553]: I will note it is still a question under discussion if this should be adjusted here or in OPD so this may change in the future but for current ts_imsim use we have decided to move forward with this solution for the time being.

Comment (jbkalmbach) [reply python/lsst/ts/imsim/closedLoopTask.py:553]: And also Josh Meyers will add a section to [sitcom-tn.](https://sitcomtn-003.lsst.io/) on the batoid coordinate system that will document these differences more formally.
